homeassistant/helpers/device_registry.py

```diff
@@ -904,6 +904,7 @@ def async_update_device(
         if not new_values:
             return old
 
+        self.hass.verify_event_loop_thread("async_update_device")
```

Comment:
Check is at the point we know we are going to write to minimize the overhead since its the write that could corrupt the internal state